googletest/include/gtest/gtest.h

```diff
@@ -1751,6 +1751,11 @@ class WithParamInterface {
   typedef T ParamType;
   virtual ~WithParamInterface() {}
 
+  // Returns true if the test is called with a parameter,
+  // i.e. within TEST_P.
+  // This function is helpful for test fixture member functions.
+  bool HasParam() const { return parameter_; }
```

Comment:
I'm a little confused about this. When wouldn't the caller know the kind of test fixture they're using?